src/demo/canvasCache.js

```diff
@@ -39,12 +39,8 @@ export default class CanvasCache {
       cacheHit = false;
       canvasObject.key = key;
       const ctx = canvasObject.canvas.getContext('2d');
-      ctx.clearRect(
-        0,
-        0,
-        canvasObject.canvas.width,
-        canvasObject.canvas.height
-      );
+      ctx.restore();
+      ctx.save();
```

Comment:
We should actually explicitly `save` at the beginning of something and do a matching `restore` at the end since these are managing a stack.